### PR TITLE
Input output refactor

### LINKED ISSUE #47: Implement easier to handle input and output of simulations
- Inputs (scenario) are written in the `simulation_run` file. We could have a config file (YAML) to easily wrote the different scenarii we want to run.
- Outputs are written in a report file, one for each scenario. For many scenarri it will be a pain to have a grasp of all the results. We could have ONE dataframe that sums up all the results in single object. And then writes it as a cvs for easy parsing.

Based on experience lauchning simulations #46

### PR DESCRIPTION
**Pull request by RomainGoussault**
Fixes #47

### CODE REVIEW

**Comment by RomainGoussault** [comment]
TODO:

- [ ] add argument parser to choose which config file to run
- [ ] handle more scenario parameters in config file
- [ ] generate more robut scenario name

**Comment by RomainGoussault** [comment]
You can filter out the .txt files and the .npy files when reviewing !

**Comment by natct10** [comment]
> You can filter out the .txt files and the .npy files when reviewing !

Thx @RomainGoussault , the diff is huuuge in this PR, I will need some time to test & review!

File: config.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+   samples_split_option: 'Random'
+   testset_option: 'Centralised'
+ - nodes_counts: 3
+   amounts_per_node: [0.9, 0.05, 0.05] 
+   samples_split_option: 'Random'
+   testset_option: 'Centralised'
+ - nodes_counts: 2
+   amounts_per_node: [0.9, 0.1] 
+   samples_split_option: 'Random'
+   testset_option: 'Centralised'
```
Comment (natct10) [inline config.yml]: Can you please add a new blank line at the end of the document (to reduce the diff)?

File: config_quick_test.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+   samples_split_option: 'Random'
+   testset_option: 'Centralised'
+ - nodes_counts: 3
+   amounts_per_node: [0.9, 0.05, 0.05] 
+   samples_split_option: 'Random'
+   testset_option: 'Centralised'
+ - nodes_counts: 2
+   amounts_per_node: [0.9, 0.1] 
+   samples_split_option: 'Random'
+   testset_option: 'Centralised'
```
Comment (natct10) [inline config_quick_test.yml]: Please add a new line :)

File: scenario.py [hunk 3]
```diff
@@ -11,12 +11,13 @@
...
 import matplotlib.pyplot as plt
 import uuid
+import pandas as pd
 
 from node import Node
 
 
 class Scenario:
-    def __init__(self, is_quick_demo=False):
+    def __init__(self, params, experiment_path):
```
Comment (natct10) [inline scenario.py]: :+1:

File: scenario.py [hunk 4]
```diff
@@ -63,12 +67,37 @@ def __init__(self, is_quick_demo=False):
...
+            + " "
+            + str(self.amounts_per_node)
+        )
+
+        self.save_folder = experiment_path / self.scenario_name
 
+        self.save_folder.mkdir(parents=True, exist_ok=True)
+
+        is_quick_demo = False
         if is_quick_demo:
```
Comment (Thomas-Galtier) [inline scenario.py]: line 100 to 103 are dead code ?

Comment (RomainGoussault) [reply scenario.py]: Yes, I changed that !

File: utils.py [hunk 5]
```diff
@@ -46,3 +53,62 @@ def preprocess_input(x):
...
+    full_experiment_name = cfg["experiment_name"] + "_" + now_str
+    experiment_path = Path.cwd() / "experiments" / full_experiment_name
+
+    # Check if experiment folder already exists
+    if experiment_path.exists():
+        logger.warning(f"Experiment folder, {experiment_path} already exists")
+        new_experiment_name = Path(str(experiment_path) + "_bis")
+        experiment_path = Path.cwd() / "experiments" / new_experiment_name
+        logger.warning(f"Experiment folder has been renamed to: {experiment_path}")
+
```
Comment (natct10) [inline utils.py]: :+1:

**Pull request merged**